### PR TITLE
chore: add forester pubkey and additional bytes to meta data

### PR DESCRIPTION
**Pull request by ananas-block**
Issue:
- only light foresters can forester Merkle trees
- additional rent added to a Merkle tree for rollover fee calculation is not preserved at rollover additional rent 
Solution:
- guard additional rent and fix later
- add additional forester pubkey to meta data (will be used in registry program)

### CODE REVIEW

File: programs/account-compression/src/processor/initialize_nullifier_queue.rs [hunk 1]
```diff
@@ -28,12 +29,14 @@ pub fn process_initialize_nullifier_queue<'a, 'b, 'c: 'info, 'info>(
             // rollover that is defined in the Merkle tree is calculated to
             // rollover the tree, queue and cpi context account.
             rollover_fee: 0,
+            additional_bytes: 0,
```
Comment (SwenSchaeferjohann) [inline programs/account-compression/src/processor/initialize_nullifier_queue.rs:32]: let's add comment that marks it as placeholder

File: test-utils/src/test_env.rs [hunk 2]
```diff
@@ -508,8 +510,8 @@ pub async fn create_state_merkle_tree_and_queue_account<R: RpcConnection>(
             merkle_tree_config.clone(),
             queue_config.clone(),
             program_owner,
+            forester,
             index,
```
Comment (SwenSchaeferjohann) [inline test-utils/src/test_env.rs:514]: this seems off? 

function signature says: 

pub fn create_initialize_merkle_tree_instruction(
    payer: Pubkey,
    registered_program_pda: Option<Pubkey>,
    merkle_tree_pubkey: Pubkey,
    nullifier_queue_pubkey: Pubkey,
    state_merkle_tree_config: StateMerkleTreeConfig,
    nullifier_queue_config: NullifierQueueConfig,
    program_owner: Option<Pubkey>,
    index: u64,
    additional_rent: u64,
)

File: programs/account-compression/src/state/access.rs [hunk 3]
```diff
@@ -7,13 +7,15 @@ pub struct AccessMetadata {
     pub owner: Pubkey,
     /// Program owner of the Merkle tree. This will be used for program owned Merkle trees.
     pub program_owner: Pubkey,
+    pub forester: Pubkey,
```
Comment (SwenSchaeferjohann) [inline programs/account-compression/src/state/access.rs:10]: add comment that explains the reasoning for placeholder pubkey. can be done in follow up pr

File: js/compressed-token/src/idl/light_compressed_token.ts [hunk 4]
```diff
@@ -753,6 +753,18 @@ export type LightCompressedToken = {
                         ];
                         type: 'publicKey';
                     },
+                    {
+                        name: 'forester';
+                        docs: [
+                            'Optional priviledged forester pubkey, can be set for custom Merkle trees',
```
Comment (ananas-block) [inline js/compressed-token/src/idl/light_compressed_token.ts:759]:
```suggestion
                            'Optional privileged forester pubkey, can be set for custom Merkle trees',
```

File: js/compressed-token/src/idl/light_compressed_token.ts [hunk 5]
```diff
@@ -1478,6 +1486,10 @@ export type LightCompressedToken = {
                     },
                     {
                         name: 'additionalBytes';
+                        docs: [
+                            'Placeholder for bytes of additional accounts which are tied to the',
+                            'Merkle trees operation and need to be rolled over as well.',
```
Comment (SwenSchaeferjohann) [inline js/compressed-token/src/idl/light_compressed_token.ts:1491]: Merkle tree's

File: js/compressed-token/src/idl/light_compressed_token.ts [hunk 6]
```diff
@@ -3041,6 +3061,10 @@ export const IDL: LightCompressedToken = {
                     },
                     {
                         name: 'additionalBytes',
+                        docs: [
+                            'Placeholder for bytes of additional accounts which are tied to the',
+                            'Merkle trees operation and need to be rolled over as well.',
```
Comment (SwenSchaeferjohann) [inline js/compressed-token/src/idl/light_compressed_token.ts:3066]: same

**Review by SwenSchaeferjohann — APPROVED** [review]

**Pull request merged**